File: 1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int num = nums.size();//返回容器中数据个数
        vector<int> rnum;
        vector<int> frnum;
        for (int i=0;i<num;i++){
            for(int j=0;j<num;j++){
                if (nums.at(i)+nums.at(j)==target)
                {
                    /* code */
                    rnum.push_back(i);
                    rnum.push_back(j);
                    break;
                }
                
            }
        }
        for (int i=0;i<rnum.size();i=i+2){
            if(rnum.at(i)!=rnum.at(i+1)){
                frnum.push_back(rnum.at(i));
                frnum.push_back(rnum.at(i+1));
                break;
            }
        }
        return frnum;
    }
};
// @lc code=end

File: 1.两数之和.java
import java.util.Map;

/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map <Integer,Integer> map = new HashMap<>();
        for (int i = 0;i<nums.length;i++){
            int temp = target -nums[i];
            if (map.containsKey(temp)){
                return new int []{map.get(temp),i};
            }
            map.put(nums[i],i);
        }
        return new int[] {-1,-1};
    }
}
// @lc code=end

